### PR TITLE
Overhaul incoming webhook docs

### PR DESCRIPTION
**Pull request by eseliger**
This PR does the following:

- Remove wording about lack of Bitbucket cloud secrets support
- Show all events to configure in a single list, so admins are more likely to enable all webhooks at once
- Slight wording tweaks and formatting to make the page feel a bit nicer
- Add some documentation about workspace level bitbucket cloud webhooks

Closes SRC-401

Closes SRC-392

Test plan:

Review.

### CODE REVIEW

**Comment by eseliger** [comment]
* **#410** <a href="https://app.graphite.dev/github/pr/sourcegraph/docs/410?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @eseliger and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by ggilmore — APPROVED** [review]

File: docs/admin/config/webhooks/incoming.mdx [hunk 1]
```diff
@@ -1,29 +1,43 @@
 # Incoming webhooks
 
-Incoming webhooks can be configured on a Sourcegraph instance in order to receive webhook events from code hosts. This allows Sourcegraph to react more quickly to events that occur outside the instance instead of polling for changes.
+Incoming webhooks can be configured on a Sourcegraph instance in order to receive events from code hosts. This allows Sourcegraph to react more quickly to events that occur outside the instance, and decrease the amount of polling for changes required.
```
Comment (MaedahBatool) [inline docs/admin/config/webhooks/incoming.mdx:3]:
```suggestion
A Sourcegraph instance can be configured to receive events from code hosts via incoming webhooks. This allows Sourcegraph to react more quickly to events outside the instance and decreases the polling required for changes.
```
Maybe a more refined version?

File: docs/admin/config/webhooks/incoming.mdx [hunk 2]
```diff
@@ -34,136 +48,203 @@ The incoming webhook will be configured to accept events from a specific code ho
...
-- `Memberships`
-- `Organizations`
-- `Repositories`
-- `Teams`
-
-When one of these events occur, a permissions sync will trigger for the relevant user or repository.
+### GitLab
 
-> NOTE: Permission changes can take a few seconds to reflect on GitHub. To prevent syncing permissions before the change reflects on GitHub, the permissions sync will only occur 10 seconds after the relevant event is received.
+GitLab webhooks can be installed at the repository or the project level. We strongly recommend using project level webhooks, to decrease the amount of work needed to add webhooks to every repository, and ensure new repositories have the webhooks added as well.
```
Comment (MaedahBatool) [inline docs/admin/config/webhooks/incoming.mdx:101]:
```suggestion
GitLab webhooks can be installed at the repository or project level. We strongly recommend using project-level webhooks to decrease the amount of work needed to add webhooks to every repository and ensure that new repositories have them added as well.
```

File: docs/admin/config/webhooks/incoming.mdx [hunk 3]
```diff
@@ -34,136 +48,203 @@ The incoming webhook will be configured to accept events from a specific code ho
...
 1. Confirm that the new webhook is listed under **All webhooks** with a timestamp in the **Last successful** column.
 
-Done! Sourcegraph will now receive webhook events from Bitbucket Server / Bitbucket Data Center and use them to sync pull request events, used by [batch changes](/batch_changes/), faster and more efficiently.
+Done! Sourcegraph will now receive webhook events to sync data with lower latency and more efficiently.
 
-#### Code push
+### Bitbucket cloud
 
-Follow the same steps as above, but ensure you tick the `Push` option. If asked for a specific event, use `repo:refs_changed`.
+Bitbucket Cloud webhooks can be installed at the repository, or at the workspace level. While we recommend installing them at the workspace level, the setup for them on the Bitbucket side [is unfortunately a bit more involved](https://community.atlassian.com/t5/Bitbucket-questions/How-can-you-create-Workspace-webhooks/qaq-p/2200142).
```
Comment (MaedahBatool) [inline docs/admin/config/webhooks/incoming.mdx:159]:
```suggestion
Bitbucket Cloud webhooks can be installed at the repository or workspace level. While we recommend installing them at the workspace level, the setup for them on the Bitbucket side [is unfortunately a bit more involved](https://community.atlassian.com/t5/Bitbucket-questions/How-can-you-create-Workspace-webhooks/qaq-p/2200142).
```

**Review by MaedahBatool — APPROVED** [review]
Small tweaks. Rest looks good :)

**Pull request merged**